stac_fastapi/core/stac_fastapi/core/types/core.py

```diff
@@ -160,9 +160,25 @@ async def create_catalog(
         """
         ...
 
+    @abc.abstractmethod
+    async def create_super_catalog(
+        self, catalog: stac_types.Catalog, **kwargs
+    ) -> Optional[Union[stac_types.Catalog, Response]]:
```

Comment:
minor point, but the preferred way to declare a union is now `Optional[stac_types.Catalog | Response]`, mostly for brevity. https://fastapi.tiangolo.com/it/python-types/#union. No need to update this as `Union` still perfectly acceptable.

Reply:
Updated as suggested